aten/src/ATen/native/Unique.cpp

```diff
@@ -275,7 +279,7 @@ unique_dim_consecutive_cpu(const Tensor& self, const int64_t dim, const bool ret
 
 std::tuple<Tensor, Tensor, Tensor>
 unique_consecutive_cpu(const Tensor& self, const bool return_inverse, const bool return_counts, c10::optional<int64_t> dim) {
-  if (!dim.has_value()) {
+  if (!dim.has_value() || (dim.value() == 0 && self.sizes().size() == 1)) {
```

Comment:
Without this check, the result would be

```
torch.unique_consecutive(dim=0) time: 16.88133478164673
torch.unique_consecutive() time: 0.024564027786254883
torch.unique_consecutive(dim=0) time: 28.210450410842896
torch.unique_consecutive(dim=1) time: 0.30884695053100586
```

As you could see, this check slightly slows down `torch.unique_consecutive(dim=1)`, but it could also speed up `torch.unique_consecutive(dim=0)` very much.
Hence, I still leave it here.

Reply:
How can this check affect `dim=1` case? With or without this check `dim=1` case should go to `unique_dim_consecutive_cpu` in line 287. 5x slowdown (from 0.3 to 1.6) seems significant, why is your algo slower than the previous one for `dim=1`?

Reply:
Maybe some CPU branching issue?

If I tried loop the last case 5 times, the result become
```
torch.unique_consecutive(dim=0) time: 0.06868529319763184
torch.unique_consecutive() time: 0.06931805610656738
torch.unique_consecutive(dim=0) time: 26.346323251724243
torch.unique_consecutive(dim=1) time: 1.2163631916046143
torch.unique_consecutive(dim=1) time: 0.2443101406097412
torch.unique_consecutive(dim=1) time: 0.24067211151123047
torch.unique_consecutive(dim=1) time: 0.25301051139831543
torch.unique_consecutive(dim=1) time: 0.24738121032714844
```

I will try `torch.utils.benchmark` or something else to get a more thorough analysis if possible.

Reply:
Ah ok, it seems like a warm-up issue then, but please post torch.utils.benchmark timings here, so that we have a record.

Reply:
The script I test [test_1d.py](https://gist.github.com/oToToT/ba1a8951d55628354a03ac18d3871d2c#file-test_1d-py), [test_2d.py](https://gist.github.com/oToToT/ba1a8951d55628354a03ac18d3871d2c#file-test_2d-py).

Result on be091950: [1d result](https://gist.github.com/oToToT/ba1a8951d55628354a03ac18d3871d2c#file-1d_result_be091950-txt), [2d result](https://gist.github.com/oToToT/ba1a8951d55628354a03ac18d3871d2c#file-2d_result_be091950-txt)

Result on 89e66488 w/ `(dim.value() == 0 && self.sizes().size() == 1)`: [1d result](https://gist.github.com/oToToT/ba1a8951d55628354a03ac18d3871d2c#file-1d_result_89e66488-txt), [2d result](https://gist.github.com/oToToT/ba1a8951d55628354a03ac18d3871d2c#file-2d_result_89e66488-txt)

Result on 89e66488 w/o `(dim.value() == 0 && self.sizes().size() == 1)`: [1d result](https://gist.github.com/oToToT/ba1a8951d55628354a03ac18d3871d2c#file-1d_result_89e66488_no_check-txt), [2d result](https://gist.github.com/oToToT/ba1a8951d55628354a03ac18d3871d2c#file-2d_result_89e66488_no_check-txt)

Summary:
For 1d case, >100x more faster on `dim=0` case with `(dim.value() == 0 && self.sizes().size() == 1)`.
For 2d case, there is no significant impact with or without `(dim.value() == 0 && self.sizes().size() == 1)`.
In general, the new implementation (89e66488) is 3x more faster than the original implementation (be091950).